hail_search/queries/mito.py

```diff
@@ -118,6 +122,177 @@ class MitoHailTableQuery(BaseHailTableQuery):
+            return super()._import_and_filter_entries_ht(project_guid, num_families, project_sample_type_data, **kwargs)
+
+        self._has_both_sample_types = True
+        entries = {}
+        for sample_type in sample_types:
+            ht, sample_data = self._load_family_or_project_ht(
+                num_families, project_guid, project_sample_type_data, sample_type, **kwargs
+            )
+            entries[sample_type] = (ht, sample_data)
+
+        return self._filter_entries_ht_both_sample_types(
+            *entries[SampleType.WES.value], *entries[SampleType.WGS.value], **kwargs
+        )
+
+    def _import_and_filter_multiple_project_hts(
+        self, project_samples: dict, n_partitions=MAX_PARTITIONS, **kwargs
+    ) -> tuple[hl.Table, hl.Table]:
+        sample_types = set()
+        for sample_dict in project_samples.values():
+            sample_types.update(sample_dict.keys())
+        if len(sample_types) == 1:
+            return super()._import_and_filter_multiple_project_hts(project_samples, n_partitions, **kwargs)
+
+        self._has_both_sample_types = True
+        entries = self._load_project_hts_both_sample_types(project_samples, n_partitions, **kwargs)
+
+        filtered_project_hts = []
+        filtered_comp_het_project_hts = []
+        for entry in entries:
+            wes_ht, wes_project_samples = entry[SampleType.WES.value]
+            wgs_ht, wgs_project_samples = entry[SampleType.WGS.value]
+            ht, comp_het_ht = self._filter_entries_ht_both_sample_types(
+                wes_ht, wes_project_samples, wgs_ht, wgs_project_samples,
+                is_merged_ht=True, **kwargs
+            )
+            if ht is not None:
+                filtered_project_hts.append(ht)
+            if comp_het_ht is not None:
+                filtered_comp_het_project_hts.append(comp_het_ht)
+
+        return self._merge_filtered_hts(filtered_comp_het_project_hts, filtered_project_hts, n_partitions)
+
+    def _load_project_hts_both_sample_types(
+        self, project_samples: dict, n_partitions: int, **kwargs
+    ) -> list[dict[str, tuple[hl.Table, dict]]]:
+        all_project_hts = []
+        project_hts = defaultdict(list)
+        sample_data = defaultdict(dict)
+
+        for project_guid, project_sample_type_data in project_samples.items():
+            for sample_type, family_sample_data in project_sample_type_data.items():
+                project_ht = self._read_project_data(
+                    family_sample_data, project_guid, sample_type, project_hts[sample_type], sample_data[sample_type]
+                )
+                if project_ht is None:
+                    continue
+
+            # Merge both WES and WGS project_hts when either of their lengths reaches the chunk size
+            if len(project_hts[SampleType.WES.value]) >= HT_CHUNK_SIZE or len(project_hts[SampleType.WGS.value]) >= HT_CHUNK_SIZE:
```

Comment:
I mean the issue with this is it could be pretty unbalanced - you could have 64 WES projects and only 2 WGS projects but you are still going to do the merge operation on those 2 WGS, when in fact maybe you only have a small number of WGS tables total. 

I think what would make this a bit neater is if you actually differentiated between projects with multiple sample types and those with one. So before you call this function in `_import_and_filter_multiple_project_hts` when you are iterating through `project_samples` put all the projects with only 1 sample type in one list and all those with 2 sample types in the other. Then use `super()` to filter and merge all the single sample type projects, and call this helper only with projects that have both sample types. This means that here the number of WES and WGS tables will always be identical, so when you do this merge step it will be as balanced as possible and so you will minimize theoverall number of merges

Reply:
I did this. I also refactored `_load_project_hts()` on the base class as best I could so that we aren't duplicating the chunk size checking logic. Let me know what you think.